src/main/java/seedu/bookmark/model/wordstore/WordStore.java

```diff
@@ -0,0 +1,46 @@
+package seedu.bookmark.model.wordstore;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class WordStore {
+
+    /**
+     * Adds words from a given list into the Store
+     * @param words
+     */
+    public abstract void addWords(List<String> words);
```

Comment:
Instead of having all these methods being abstract, they can actually just be implemented inside `WordStore` itself and inherited by each of your different types of WordStores (TagWordStore, NameWordStore, etc).  Currently, there is a lot of repetition since each of the individual types of WordStores implement their own `addWords()`, `deleteWords()`, etc but their logic is exactly the same.

Reply:
That's a good point, I have changed it. Thanks!